src/SSHDebugPS/UI/ViewModels/ContainerPickerViewModel.cs

```diff
@@ -211,7 +244,9 @@ public IContainerInstance SelectedContainerInstance
             }
             set
             {
-                if (!(_selectedContainerInstance != null && _selectedContainerInstance.Name.Equals(value.Name, StringComparison.Ordinal)))
+                // checking cases that they are not equal
+                if (((_selectedContainerInstance != null) && (value == null || !string.Equals(_selectedContainerInstance.Name, value.Name, StringComparison.Ordinal)))
+                    || (_selectedContainerInstance == null && value != null))
```

Comment:
Now that `IContainerInstance` implements `IEquatable<T>`, can you use that here to simplify this check?